### PR TITLE
create post route, add business logic for adding post BE

### PR DESCRIPTION
**Pull request by rostyk1331**

### CODE REVIEW

File: server/src/controllers/post.controllers/create.post.ts [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+import { uploadImage } from '../../helpers/uploadImage';
+import { IPostModel } from '../../models/post.model';
+
+const awsConfig = {
+    bucket: 'jsgram-post-images1',
+    acl: 'public-read',
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY_2,
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID_2,
+    region: process.env.AWS_REGION_2,
+    fileSize: 1024 * 1024 * 4,
```
Comment (InnaKutsik) [inline server/src/controllers/post.controllers/create.post.ts]: move to constant

File: server/src/controllers/profile.controllers/handlePhoto.ts [hunk 2]
```diff
@@ -1,44 +1,17 @@
...
-const s3 = new aws.S3();
-
-const fileFilter = (req: Request, file: Express.Multer.File,
-                    cb: (error: Error | null, acceptFile: boolean) => void): void => {
-    if (file.mimetype === 'image/jpeg' || file.mimetype === 'image/png') {
-        cb(null, true);
-    } else {
-        cb(new Error('Please, choose an image in a JPEG or PNG format.'), false);
-    }
+    fileSize: 1024 * 1024 * 2,
```
Comment (InnaKutsik) [inline server/src/controllers/profile.controllers/handlePhoto.ts]: move to constant

File: server/src/controllers/post.controllers/create.post.ts [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+            const tags = req.body.description.match(hashtagRegex);
+
+            const newPost = new Post({
+                description,
+                imgPath,
+                tags,
+                author: author.id,
+            });
+
+            newPost.save((error: Error, post: IPostModel) => {
```
Comment (InnaKutsik) [inline server/src/controllers/post.controllers/create.post.ts]: move to bd requests folders and import here

Comment (rostyk1331) [reply server/src/controllers/post.controllers/create.post.ts]: Fixed.

File: server/src/controllers/post.controllers/create.post.ts (hunk 1 above)
Comment (rostyk1331) [reply server/src/controllers/post.controllers/create.post.ts]: Done.

File: server/src/controllers/profile.controllers/handlePhoto.ts (hunk 2 above)
Comment (rostyk1331) [reply server/src/controllers/profile.controllers/handlePhoto.ts]: Done.

File: server/src/db.requests/addPost.request.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export const addPost = (author: IUserModel, description: string, imgPath: string, tags: string[]): IPostModel => {
+    const newPost = new Post({
+        description,
+        imgPath,
+        tags,
+        author: author.id,
+    });
+
+    newPost.save((error: Error, post: IPostModel) => {
```
Comment (InnaKutsik) [inline server/src/db.requests/addPost.request.ts]: Pls move db query folder

File: server/src/db.requests/addPost.request.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        if (error) {
+            throw new Error('Can not create new post');
+        }
+        author.posts.push(post.id);
+        author.save();
```
Comment (InnaKutsik) [inline server/src/db.requests/addPost.request.ts]: Post is not result of save with id?

**Pull request merged**